Src/EnvironmentMapEffect.cpp

```diff
@@ -118,24 +134,42 @@ namespace
+    #include "Shaders/Compiled/EnvironmentMapEffect_PSEnvMapSpherePixelLightingFresnelNoFog.inc"
+
+    #include "Shaders/Compiled/EnvironmentMapEffect_PSEnvMapDualParabolaPixelLighting.inc"
+    #include "Shaders/Compiled/EnvironmentMapEffect_PSEnvMapDualParabolaPixelLightingNoFog.inc"
+    #include "Shaders/Compiled/EnvironmentMapEffect_PSEnvMapDualParabolaPixelLightingFresnel.inc"
+    #include "Shaders/Compiled/EnvironmentMapEffect_PSEnvMapDualParabolaPixelLightingFresnelNoFog.inc"
 #endif
 }
 
 
 template<>
 const ShaderBytecode EffectBase<EnvironmentMapEffectTraits>::VertexShaderBytecode[] =
 {
-    { EnvironmentMapEffect_VSEnvMap,                  sizeof(EnvironmentMapEffect_VSEnvMap)                  },
-    { EnvironmentMapEffect_VSEnvMapFresnel,           sizeof(EnvironmentMapEffect_VSEnvMapFresnel)           },
-    { EnvironmentMapEffect_VSEnvMapOneLight,          sizeof(EnvironmentMapEffect_VSEnvMapOneLight)          },
-    { EnvironmentMapEffect_VSEnvMapOneLightFresnel,   sizeof(EnvironmentMapEffect_VSEnvMapOneLightFresnel)   },
-    { EnvironmentMapEffect_VSEnvMapPixelLighting,     sizeof(EnvironmentMapEffect_VSEnvMapPixelLighting)     },
-
-    { EnvironmentMapEffect_VSEnvMapBn,                sizeof(EnvironmentMapEffect_VSEnvMapBn)                },
-    { EnvironmentMapEffect_VSEnvMapFresnelBn,         sizeof(EnvironmentMapEffect_VSEnvMapFresnelBn)         },
-    { EnvironmentMapEffect_VSEnvMapOneLightBn,        sizeof(EnvironmentMapEffect_VSEnvMapOneLightBn)        },
-    { EnvironmentMapEffect_VSEnvMapOneLightFresnelBn, sizeof(EnvironmentMapEffect_VSEnvMapOneLightFresnelBn) },
-    { EnvironmentMapEffect_VSEnvMapPixelLightingBn,   sizeof(EnvironmentMapEffect_VSEnvMapPixelLightingBn)   },
+    { EnvironmentMapEffect_VSEnvMap,                   sizeof(EnvironmentMapEffect_VSEnvMap)                   },
+    { EnvironmentMapEffect_VSEnvMapFresnel,            sizeof(EnvironmentMapEffect_VSEnvMapFresnel)            },
+    { EnvironmentMapEffect_VSEnvMapOneLight,           sizeof(EnvironmentMapEffect_VSEnvMapOneLight)           },
+    { EnvironmentMapEffect_VSEnvMapOneLightFresnel,    sizeof(EnvironmentMapEffect_VSEnvMapOneLightFresnel)    },
+    { EnvironmentMapEffect_VSEnvMapPixelLighting,      sizeof(EnvironmentMapEffect_VSEnvMapPixelLighting)      },
+
+    { EnvironmentMapEffect_VSEnvMapBn,                 sizeof(EnvironmentMapEffect_VSEnvMapBn)                 },
+    { EnvironmentMapEffect_VSEnvMapFresnelBn,          sizeof(EnvironmentMapEffect_VSEnvMapFresnelBn)          },
+    { EnvironmentMapEffect_VSEnvMapOneLightBn,         sizeof(EnvironmentMapEffect_VSEnvMapOneLightBn)         },
+    { EnvironmentMapEffect_VSEnvMapOneLightFresnelBn,  sizeof(EnvironmentMapEffect_VSEnvMapOneLightFresnelBn)  },
+    { EnvironmentMapEffect_VSEnvMapPixelLightingBn,    sizeof(EnvironmentMapEffect_VSEnvMapPixelLightingBn)    },
+
+#if defined(_XBOX_ONE) && defined(_TITLE)
```

Comment:
The dual-paraboloid  PIxel Shaders shaders use Shader Model 4.0. Therefore, I need a SM4 Vertex Shader to match them. For Xbox, they are all Shader Model 5 so I can reuse the same VS.